task6/index.js

```diff
@@ -0,0 +1,364 @@
+            validUntil: new Date('2021-07-30T24:00:00'),
+            discount: '10%',
+            hashTags: ['Красота и здоровье', 'Прочие услуги']
+        },
+        {
+            id: '18',
+            description: 'Скидки до 70% в магазинах Kari и Kari KIDS!',
+            createdAt: new Date('2021-03-01T11:00:00'),
+            link: 'https://by.kari.com/by/',
+            vendor: 'Kari',
+            photoLink: 'https://trk-pilot.ru/wp-content/uploads/2018/04/Logo_Kari_newest.jpg',
+            validUntil: new Date('2021-07-01T24:00:00'),
+            discount: '70%',
+            hashTags: ['Прочие услуги']
+        },
+        {
+            id: '19',
+            description: 'Скидки в магазинах 21vek',
+            createdAt: new Date('2021-03-08T11:00:00'),
+            link: 'https://www.21vek.by/',
+            vendor: '21vek',
+            photoLink: 'http://i.mycdn.me/i?r=AzGBqNaF5OQp2lMpnhRx4DEF7OiGMi6UuMkBd-_FgVClksjukuIIGj_LwIulBvzyPuY',
+            validUntil: new Date('2021-08-08T24:00:00'),
+            discount: '40%',
+            hashTags: ['Прочие услуги']
+        },
+        {
+            id: '20',
+            description: 'Обучение иностранным языкам в "SOL Minsk"',
+            createdAt: new Date('2021-03-20T11:00:00'),
+            link: 'https://sol.by/',
+            vendor: 'SOL',
+            photoLink: 'https://enguide.by/s/public/upload/images/e6ce/8bd6/f526/3246/470a/4b7d/f2ed/7df0.jpeg',
+            validUntil: new Date('2021-08-21T24:00:00'),
+            discount: '20%',
+            hashTags: ['Образовательные услуги']
+        }
+    ]
+
+    function getAds(skip = 0, top = 10, filterConfig = undefined) {
+        if (typeof skip != 'number' || typeof top != 'number') {
+            return 'Error'
+        }
+        let elem = adList;
+        if (filterConfig) {
+            for (let item in filterConfig) {
+                if (item === 'createdAt') {
+                    elem = elem.filter(item => item.createdAt <= filterConfig.createdAt);
+                }
+                if (item === 'validUntil') {
+                    elem = elem.filter(item => item.validUntil >= filterConfig.validUntil);
+                }
+                if (item === 'vendor') {
+                    elem = elem.filter(item => item.vendor === filterConfig.vendor);
+                }
+                else {
+                    return 'Error'
+                }
+            }
+        }
+        elem.sort(function (a, b) {
+            return a.createdAt - b.createdAt;
+        });
+        return elem.slice(skip, skip + top);
+    }
+
+    console.log(getAds(0, 10))
+    console.log(getAds(10, 10))
+    console.log(getAds('3'))
+    console.log(getAds(0, 4, {vendor: 'Анна Петрачкова'}))
+    console.log(getAds(0, 7, {validUntil: new Date('2021-05-01T24:00:00')}))
+
+
+    function getAd(id) {
+        return adList.filter(item => item.id == id)
```

Comment:
`Array.filter` returns array of adItems, you need to return single item, so please use `Array.find`.